ivadomed/loader/loader.py

```diff
@@ -437,30 +445,44 @@ def __getitem__(self, index):
         """
         coord = self.indexes[index]
         input_img, gt_img = self.handlers[coord['handler_index']].get_pair_data()
-        data_shape = gt_img[0].shape
-        seg_pair_slice = self.handlers[coord['handler_index']].get_pair_slice(coord['handler_index'])
+        data_shape = input_img[0].shape
+        seg_pair_slice = self.handlers[coord['handler_index']].get_pair_metadata(coord['handler_index'])
         data_dict = {
             'input': input_img,
-            'gt': gt_img
+            'gt': gt_img,
+            'input_metadata': seg_pair_slice['input_metadata'],
+            'gt_metadata': seg_pair_slice['gt_metadata']
         }
-
-        for idx in range(len(data_dict['input'])):
-            data_dict['input'][idx] = data_dict['input'][idx][coord['x_min']:coord['x_max'],
-                                      coord['y_min']:coord['y_max'],
-                                      coord['z_min']:coord['z_max']]
-
-        for idx in range(len(data_dict['gt'])):
-            data_dict['gt'][idx] = data_dict['gt'][idx][coord['x_min']:coord['x_max'],
-                                   coord['y_min']:coord['y_max'],
-                                   coord['z_min']:coord['z_max']]
-
-        data_dict['input_metadata'] = seg_pair_slice['input_metadata']
-        data_dict['gt_metadata'] = seg_pair_slice['gt_metadata']
         for idx in range(len(data_dict["input"])):
             data_dict['input_metadata'][idx]['data_shape'] = data_shape
+
         if self.transform is not None:
             data_dict = self.transform(data_dict)
-        return data_dict
+
+        shape_x = coord["x_max"] - coord["x_min"]
+        shape_y = coord["y_max"] - coord["y_min"]
+        shape_z = coord["z_max"] - coord["z_min"]
+
+        subvolumes = {
+            'input': torch.zeros(data_dict['input'].shape[0], shape_x, shape_y, shape_z),
+            'gt': torch.zeros(data_dict['input'].shape[0], shape_x, shape_y, shape_z),
+            'input_metadata': seg_pair_slice['input_metadata'],
+            'gt_metadata': seg_pair_slice['gt_metadata']
+        }
+
+        for idx in range(len(data_dict['input'])):
+            subvolumes['input'] = data_dict['input'][:,
+                                  coord['x_min']:coord['x_max'],
+                                  coord['y_min']:coord['y_max'],
+                                  coord['z_min']:coord['z_max']]
```

Comment:
I only had a quick look: Is `idx` not used?

Reply:
Updated :) thanks!